### PR TITLE
KAFKA-18361: Remove PasswordEncoderConfigs

### PR DESCRIPTION
**Pull request by frankvicky**
JIRA: KAFKA-18361

> In `kfrat` mode, we do not encrypt the values in records.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — COMMENTED** [review]
@frankvicky please add the changes to upgrade.html

File: docs/upgrade.html [hunk 1]
```diff
@@ -56,6 +56,8 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
                         <li>The <code>org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerValidatorCallbackHandler</code> class was removed.
                             Please use the <code>org.apache.kafka.common.security.oauthbearer.OAuthBearerValidatorCallbackHandler</code> class instead.
                         </li>
+                        <li>The <code>org.apache.kafka.common.security.PasswordEncoderConfigs</code> class was removed.
```
Comment (chia7712) [inline docs/upgrade.html]: Could you please list the removed configs instead?

Comment (frankvicky) [reply docs/upgrade.html]: I have modified the doc as following: 
Screenshot from 2024-12-30 21-15-08

**Review by chia7712 — APPROVED** [review]

**Comment by chia7712** [comment]
the failed tests are traced by following jiras:

https://issues.apache.org/jira/browse/KAFKA-18298
https://issues.apache.org/jira/browse/KAFKA-15474
https://issues.apache.org/jira/browse/KAFKA-17676

**Pull request merged**